forte/data/readers/classification_reader.py

```diff
@@ -0,0 +1,239 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# mypy: ignore-errors
+import csv
+import importlib
+from typing import Iterator, Tuple, List
+from collections import OrderedDict
+from forte.common import Resources, ProcessorConfigError
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.data.base_reader import PackReader
+from ft.onto.base_ontology import *
+
+
+__all__ = ["ClassificationDatasetReader"]
+
+
+class ClassificationDatasetReader(PackReader):
+    r"""
+    A generic dataset reader class for classification dataset.
+    In general, it works with any classification tasks that
+    require concatenating a sequence of strings from dataset and predicting a class.
+
+
+    The user must specify the index2label mapping, data fields and subtext fields
+    and other parameters in configs based on their tasks in order to initialize
+    the dataset reader properly.
+    Index2label mapping is a dictionary with indices as keys and string labels as values,
+    it helps encoding and decoding string labels.
+    Data fields is a list of ontology paths representing data fields in the dataset.
+    They must have the same length and align with each other.
+    Subtext fields is also a list of ontology paths that represent data fields that
+     will be concatenated into input strings in the same order as the list.
+    User can select an arbitrary sequence of subtexts to concatenate
+     as long as there is at least one subtext in the sequence.
+     The number and the order of subtexts can be customized depending on the use cases.
+
+
+
+    The first line of dataset usually specifies column names of data fields,
+    and the user needs to write a list of data ontology paths.
+    For example, in Amazon review sentiment dataset, the first line specifies [content, label,
+     title].
+    By checking the actual data, except for label, we must find their corresponding ontology names
+    and its relative path in the given or customized ontology.
+    Here we define data_fields = ["label", "ft.onto.base_ontology.Title",
+     "ft.onto.ag_news.Description"]
+    And we want both Title and Description to be included in input string, so we define
+    subtext fields = [ "ft.onto.base_ontology.Title",  "ft.onto.ag_news.Description"].
+    To see a full example, please refer to
+    https://github.com/asyml/forte/examples/classification
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._index2class = None
+        self._class2index = None
+
+    def set_up(self):
+        assert self.configs.index2class is not None, (
```

Comment:
in setup you should also check other configuration values, like `data_fields` and so on

Reply:
Those checks require comparing with the input datapack so I checked under `_parse_pack` method.
https://github.com/feipenghe/forte/blob/96e14de31ddeac4d4f5201ac567d3b9076223579/forte/data/readers/classification_reader.py#L115